scionlab/tests/test_user_as_models.py

```diff
@@ -252,58 +325,40 @@ def _get_random_useras_params(seed, force_public_ip=False, force_bind_ip=False,
     """
     r = random.Random(seed)
 
-    def _randbool():
-        return r.choice([True, False])
-
-    use_vpn = kwargs.setdefault('use_vpn', _randbool())
-    # Make AP choice both with/without VPN to always consume same amount of random state
-    candidate_AP_vpn = r.choice(AttachmentPoint.objects.filter(vpn__isnull=False))
-    candidate_AP_novpn = r.choice(AttachmentPoint.objects.all())
-    if use_vpn:
-        kwargs.setdefault('attachment_point', candidate_AP_vpn)
-    else:
-        kwargs.setdefault('attachment_point', candidate_AP_novpn)
     kwargs.setdefault('owner', get_testuser())
-
-    public_ip = '172.31.0.%i' % r.randint(10, 254)
-    public_port = r.choice(range(DEFAULT_PUBLIC_PORT, DEFAULT_PUBLIC_PORT + 20))
-    if _randbool() or not use_vpn or force_public_ip:
-        kwargs.setdefault('public_ip', public_ip)
-    else:
-        kwargs.setdefault('public_ip', None)
-    kwargs.setdefault('public_port', public_port)
-
-    bind_ip = '192.168.1.%i' % r.randint(10, 254)
-    bind_port = r.choice(range(DEFAULT_PUBLIC_PORT + 1000, DEFAULT_PUBLIC_PORT + 1020))
-    if _randbool() or force_bind_ip:
-        kwargs.setdefault('bind_ip', bind_ip)
-        kwargs.setdefault('bind_port', bind_port)
-    else:
-        kwargs.setdefault('bind_ip', None)
-        kwargs.setdefault('bind_port', None)
-
     kwargs.setdefault('installation_type', r.choice((UserAS.VM, UserAS.PKG, UserAS.SRC)))
     randstr = r.getrandbits(1024).to_bytes(1024//8, 'little').decode('utf8', 'ignore')
     kwargs.setdefault('label', randstr)
 
     return kwargs
 
 
-def create_random_useras(testcase, seed, **kwargs):
+def create_and_check_random_useras(testcase, seed, aps_confs, vpn_choice, **kwargs):
     """
     Create and check UserAS with "random" parameters based on `seed`.
     Any parameters to UserAS.objects.create can be specified to override the generated values.
     """
     return create_and_check_useras(testcase,
-                                   **_get_random_useras_params(seed, **kwargs))
+                                   seed,
+                                   aps_confs,
```

Comment:
The reason why I kept this separated is because you might want to specify non-random `AttachmentConf`s. Although, looking at the code now, such a use case is actually rare.